src/server/auth-passport.js

```diff
@@ -19,90 +22,113 @@ export function setupAuth0Passport() {
-        cell: userMetadata.cell || '',
-        // eslint-disable-next-line no-underscore-dangle
-        email: req.user._json.email,
-        is_superadmin: false
-      }
-      await User.save(userData)
-      res.redirect(req.query.state || 'terms')
-      return
-    }
-    res.redirect(req.query.state || '/')
-    return
-  })]
+        const auth0Id = (req.user && (req.user.id || req.user._json.sub))
+        if (!auth0Id) {
+          throw new Error('Null user in login callback')
+        }
+        const existingUser = await User.filter({ auth0_id: auth0Id })
+
+        if (existingUser.length === 0) {
+          const userMetadata = (
+            // eslint-disable-next-line no-underscore-dangle
+            req.user._json['https://spoke/user_metadata']
+            // eslint-disable-next-line no-underscore-dangle
+              || req.user._json.user_metadata
+              || {})
+          const userData = {
+            auth0_id: auth0Id,
+            // eslint-disable-next-line no-underscore-dangle
+            first_name: userMetadata.given_name || '',
+            // eslint-disable-next-line no-underscore-dangle
+            last_name: userMetadata.family_name || '',
+            cell: userMetadata.cell || '',
+            // eslint-disable-next-line no-underscore-dangle
+            email: req.user._json.email,
+            is_superadmin: false
+          }
+          await User.save(userData)
+          res.redirect(req.query.state || 'terms')
+          return
+        }
+        res.redirect(req.query.state || '/')
+        return
+      })]
+  }
 }
 
 export function setupLocalAuthPassport() {
   const strategy = new LocalStrategy({
-    usernameField: 'email',
-    passwordField: 'auth0_id' // using the legacy fieldname for password
-  }, function (username, password, done) {
-    User.filter({ email: username }, function (err, user) {
-      if (err) { return done(err) }
-      if (!user) { return done(null, false) }
-
-        // AuthHasher.hash(password, function(err, hashed) {
-        // const passwordToSave = `${hashed.salt}|${hashed.hash}`
-        // .salt and .hash
-        // });
-      const pwFieldSplit = user.auth0_id.split('|')
+    usernameField: 'email'
+  }, wrap(async (username, password, done) => {
+    console.log('LOGIN', username, password)
+    const existingUser = await User.filter({ email: username })
+    if (existingUser.length > 0) {
+      const pwFieldSplit = existingUser[0].auth0_id.split('|')
       const hashed = {
         salt: pwFieldSplit[0],
         hash: pwFieldSplit[1]
       }
-      AuthHasher.verify(password, hashed, function (err, verified) {
-        if (verified) {
-          return done(null, false)
-        } else {
-          done(null, user)
-        }
-      })
-    })
-  }
-  )
+      AuthHasher.verify(
+        password, hashed,
+        (err, verified) => (verified
+                            ? done('Validation failed', null)
```

Comment:
I was trying to add onto your work on this PR, but didn't get too far. I'll try to get back to it soon, but the main thing I noticed is that this conditional needs to be flipped around.

Reply:
dur, thanks.

I'll need to verify that what I thought was working actually is if this was really flipped then maybe I wasn't even as far as I thought....